x-pack/plugins/index_management/__jest__/client_integration/home/data_streams_tab.test.ts

```diff
@@ -244,6 +251,25 @@ describe('Data Streams tab', () => {
           dataStreams: ['dataStream1'],
         });
       });
+
+      test('clicking index template name navigates to the index template details', async () => {
+        const {
+          actions: { clickNameAt, clickDetailPanelIndexTemplateLink },
+          findDetailPanelIndexTemplateLink,
+          component,
+          find,
+        } = testBed;
+
+        await clickNameAt(0);
+
+        const indexTemplateLink = findDetailPanelIndexTemplateLink();
+        expect(indexTemplateLink.text()).toBe('indexTemplate');
+
+        await clickDetailPanelIndexTemplateLink();
+
+        component.update();
+        expect(find('summaryTab').exists()).toBeTruthy();
```

Comment:
Could we add a check that the correct index template has been loaded?

Looking at the router path + the data-test-subj "title" of the flyout. You might have to mock the response on `httpRequestsMockHelpers.setLoadTemplateResponse()`.